### PR TITLE
Add function GetStorageProfileByHref and fix bug in type

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: https://github.com/terraform-providers/terraform-provider-vcd/pull/324

* Added function GetStorageProfileByHref - needed to make Terraform read work
* Fix bug in Vdc type

### CODE REVIEW

File: govcd/system.go [hunk 1]
```diff
@@ -687,6 +687,20 @@ func getExtension(client *Client) (*types.Extension, error) {
...
 
+// GetStorageProfileByHref fetches storage profile using provided HREF.
+func GetStorageProfileByHref(vcdClient *VCDClient, url string) (*types.VdcStorageProfile, error) {
+
+	vdcStorageProfile := &types.VdcStorageProfile{}
+
+	_, err := vcdClient.Client.ExecuteRequest(url, http.MethodGet,
+		"", "error retrieving storage profile: %s", nil, vdcStorageProfile)
+	if err != nil {
+		return &types.VdcStorageProfile{}, err
```
Comment (dataclouder) [inline govcd/system.go]: We should return `nil` when a function returning a pointer fails.

File: govcd/system_test.go [hunk 2]
```diff
@@ -452,3 +452,25 @@ func (vcd *TestVCD) Test_QueryNetworkPoolByName(check *C) {
 	check.Assert(networkPoolFound, Equals, true)
 
 }
+
+// Test getting storage profile by href and vdc client
+func (vcd *TestVCD) Test_GetStorageProfileByHref(check *C) {
+	if vcd.config.VCD.ProviderVdc.StorageProfile == "" {
+		check.Skip("Skipping test because storage profiel is not configured")
```
Comment (dataclouder) [inline govcd/system_test.go]: "profiel" -> "profile"

File: govcd/system_test.go [hunk 3]
```diff
@@ -452,3 +452,25 @@ func (vcd *TestVCD) Test_QueryNetworkPoolByName(check *C) {
...
+
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, NotNil)
+
+	adminVdc, err := adminOrg.GetAdminVDCByName(vcd.config.VCD.Vdc, false)
+	check.Assert(err, IsNil)
+	check.Assert(adminVdc, NotNil)
+
+	// Get network pool by href
```
Comment (dataclouder) [inline govcd/system_test.go]: "network pool" -> "storage profile"

File: govcd/system_test.go [hunk 4]
```diff
@@ -452,3 +452,25 @@ func (vcd *TestVCD) Test_QueryNetworkPoolByName(check *C) {
...
+	foundStorageProfile, err := GetStorageProfileByHref(vcd.client, adminVdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile[0].HREF)
+	check.Assert(err, IsNil)
+	check.Assert(foundStorageProfile, Not(Equals), types.VdcStorageProfile{})
```
Comment (dataclouder) [inline govcd/system_test.go]: Check for not nil

File: types/v56/types.go [hunk 5]
```diff
@@ -370,20 +370,20 @@ type Vdc struct {
...
+	ResourceEntities   []*ResourceEntities  `xml:"ResourceEntities,omitempty"`
+	AvailableNetworks  []*AvailableNetworks `xml:"AvailableNetworks,omitempty"`
+	Capabilities       []*Capabilities      `xml:"Capabilities,omitempty"`
+	NicQuota           int                  `xml:"NicQuota"`
+	NetworkQuota       int                  `xml:"NetworkQuota"`
+	VMQuota            int                  `xml:"VmQuota"`
+	IsEnabled          bool                 `xml:"IsEnabled"`
+	Tasks              *TasksInProgress     `xml:"Tasks,omitempty"`
+	UsedNetworkCount   int                  `xml:"UsedNetworkCount,omitempty"`
+	VdcStorageProfiles *VdcStorageProfiles  `xml:"VdcStorageProfiles"`
```
Comment (dataclouder) [inline types/v56/types.go]: Are we breaking compatibility with past releases?
Please explain what are we fixing

File: govcd/system.go (hunk 1 above)
Comment (vbauzysvmware) [reply govcd/system.go]: fixed

File: govcd/system_test.go (hunk 2 above)
Comment (vbauzysvmware) [reply govcd/system_test.go]: fixed

File: govcd/system_test.go (hunk 3 above)
Comment (vbauzysvmware) [reply govcd/system_test.go]: fixed

File: govcd/system_test.go (hunk 4 above)
Comment (vbauzysvmware) [reply govcd/system_test.go]: fixed

File: types/v56/types.go (hunk 5 above)
Comment (vbauzysvmware) [reply types/v56/types.go]:
```

**Comment by vbauzysvmware** [comment]
> Changelog entry missing ;)

Fixed

**Pull request merged**